compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -683,16 +683,16 @@ trait Implicits { self: Typer =>
         case arg :: Nil if !arg.typeSymbol.is(Param) =>
           object bindFreeVars extends TypeMap {
             var ok = true
-            def apply(t: Type) = t match {
-              case t @ TypeRef(NoPrefix, _) =>
+            def apply(t: Type) = t.dealias match {
+              case t @ TypeRef(NoPrefix, _) if !ctx.inInlineMethod =>
```

Comment:
I did not understand the qualification `!ctx.inInlineMethod`. This looks fishy where it is. The other synthesized arguments do not depend on this test, so why would `quoted.Type` be different?

Reply:
In the case below in the outer match, we have the following logic:

https://github.com/lampepfl/dotty/blob/c165638cb46e498156bffcb5ad4ad4775d11bf35/compiler/src/dotty/tools/dotc/typer/Implicits.scala#L701-L702

I conjecture the motivation is that in inline methods, type parameters are treated as constants so that they are not restricted by _phase consistency law_, just like inline term parameters.  /cc : @nicolasstucki 

More concretely, without the test, the following test case `i4514.scala` fail:

```Scala
object Foo {
  inline def foo[X](x: X): Unit = ${fooImpl('x)}
  def fooImpl[X: quoted.Type](x: X): quoted.Expr[Unit] = '{}
}
```

The  implicit search first looks for `Type[Expr[X]]`, the `mapOver` will reach `X`, and the implicit search for `Type[X]` fails.